### PR TITLE
Intro the performance_analysis module to gather perf stats by categories

### PR DESCRIPTION
**Pull request by xymus**
This modules provides simple stats (avg, min, max, count) on the execution time of categories. I use this module in my games to detect performance bottlenecks.

If anyone has an idea for a better name, I'm open to suggestions.

### CODE REVIEW

File: lib/performance_analysis.nit [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+end
+
+# Collection of statistics on many events
+class PerfMap
+	super HashMap[String, PerfEntry]
+
+	redef fun provide_default_value(key)
+	do
+		if not key isa String then super
+		assert key isa String
```
Comment (privat) [inline lib/performance_analysis.nit]: `then return super` so you can remove  the assert

File: lib/performance_analysis.nit [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+		var ts = new PerfEntry(key)
+		self[key] = ts
+		return ts
+	end
+
+	redef fun to_s
+	do
+		var lines = new Array[String]
+		for k, v in self do lines.add "* {k}: {v}"
+		return lines.join("\n")
```
Comment (privat) [inline lib/performance_analysis.nit]: Map has a join, why not using it?
`return "* " + join(": ", "\n* ")` ?

File: lib/performance_analysis.nit [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+end
+
+# Statistics on wall clock execution time of a category of events by `name`
+class PerfEntry
+
+	# Name of the category
+	var name: String
+
+	# Shortest execution time of registered events
+	var min: Float = -1.0
```
Comment (privat) [inline lib/performance_analysis.nit]: Document the -1.0 default value. why not 0.0?

**Comment by xymus** [comment]
Rerolled with @privat comments.

**Comment by privat** [comment]
+1

**Comment by R4PaSs** [comment]
:shipit:

**Pull request merged**